crypto/dsa/dsa_asn1.c

```diff
@@ -32,6 +32,15 @@ void DSA_SIG_get0(BIGNUM **pr, BIGNUM **ps, const DSA_SIG *sig)
     *ps = sig->s;
 }
 
+int DSA_SIG_set0(BIGNUM *r, BIGNUM *s, DSA_SIG *sig)
```

Comment:
I now also noticed that the getters that already exist return mutable bignums, which suggests you don't need this extra API. However that looks wrong to me and I'd prefer your approach, so hold on for a sec and let me clarify what's going on.